### ISSUE TITLE
AndOTP KeyStore error on Android 11/Pixel 4

### ISSUE THREAD

**Issue by krc** [andOTP/andOTP#847]
#### General information

* **App version: 0.9.0-play
* **App source:** Google Play
* **Android Version:**  11
* **Custom ROM:** N/A
* **Device:** Pixel 4

#### Expected result
Able to use AndOTP with the system KeyStore, or failing that, at least able to use AndOTP's built in pin/password based encryption.

**What does happen instead?**
Not able to use KeyStore, app crashes when switching to internal encryption, remains on KeyStore

#### Logcat

https://gist.github.com/krc/4aecc010cf727eb49dfe1b74e7b08632

#### Steps to reproduce
 
launch andotp
enter previously set pin
get warning about KeyStore not being usable
change settings to app encryption (pin/password)
screen flashes
get warning about KeyStore
check settings, it is still on KeyStore
change to app encryption
screen flashes, warning comes up that the app keeps stopping

**Comment by krc** [comment]
I also tried restarting the phone and uninstalling and reinstalling AndOTP. No change

**Comment by flocke** [comment]
The problem is that andOTP can not access your data because of the KeyStore error. Because of that it can not migrate to the internal encryption.

How did you get your data into andOTP on the Pixel 4? Did you import a backup directly in andOTP?

If you still have a backup you can just reset the app and set it up again with the internal encryption.

**Comment by krc** [comment]
I did a general Android data transfer via USB cable. I never imported any AndOTP data at all. 
How do I reset the app?

**Comment by krc** [comment]
Ah, I cleared cache and data in the Android Settings - Apps... must have tried to use data from the old one because of the Android data transfer?

**Comment by flocke** [comment]
Yes, the data transfer probably copied over the old data. But it can not copy the entries from the KeyStore. So you were left with an encrypted database but no keys in the KeyStore to decrypt it.
